### PR TITLE
docs(clients/java): Dead links in docs (#258)

### LINKED ISSUE #258: Dead links in documentation
We've come across some dead links. There might be more.

Pages with dead links found:
- https://docs.camunda.io/docs/product-manuals/clients/java-client/job-worker (`BackoffSupplier`, `JobWorkerBuilder`, `gRPC Technical Error Handling`, `backpressure mechanism`)
- https://docs.camunda.io/docs/product-manuals/clients/java-client-examples/job-worker-open (`Source on github`)
- other java-client-examples with `Source on github`

### PR DESCRIPTION
**Pull request by akeller**
Closes #258.

### CODE REVIEW

File: docs/product-manuals/clients/java-client/job-worker.md [hunk 1]
```diff
@@ -31,8 +31,8 @@ It waits for some time, after which it polls again for more jobs.
 This gives a Zeebe cluster some time to recover from a failure.
 In some cases, you may want to configure this backoff strategy to better fit your situation.
 
-The retry delay (i.e. the time the job worker waits after an error before the next poll for new jobs) is provided by the [`BackoffSupplier`](https://github.com/zeebe-io/zeebe/blob/develop/clients/java/src/main/java/io/zeebe/client/api/worker/BackoffSupplier.java).
-You can replace it using the `.backoffSupplier()` method on the [`JobWorkerBuilder`](https://github.com/zeebe-io/zeebe/blob/develop/clients/java/src/main/java/io/zeebe/client/api/worker/JobWorkerBuilderStep1.java).
+The retry delay (i.e. the time the job worker waits after an error before the next poll for new jobs) is provided by the [`BackoffSupplier`](https://github.com/camunda-cloud/zeebe/blob/develop/clients/java/src/main/java/io/camunda/zeebe/client/api/worker/BackoffSupplier.java).
+You can replace it using the `.backoffSupplier()` method on the [`JobWorkerBuilder`](https://github.com/zeebe-io/zeebe/blob/develop/clients/java/src/main/java/io/camunda/zeebe/client/api/worker/JobWorkerBuilderStep1.java).
```
Comment (korthout) [inline docs/product-manuals/clients/java-client/job-worker.md:35]:
```suggestion
The retry delay (i.e. the time the job worker waits after an error before the next poll for new jobs) is provided by the [`BackoffSupplier`](https://github.com/camunda-cloud/zeebe/blob/develop/clients/java/src/main/java/io/camunda/zeebe/client/api/worker/BackoffSupplier.java).
You can replace it using the `.backoffSupplier()` method on the [`JobWorkerBuilder`](https://github.com/camunda-cloud/zeebe/blob/develop/clients/java/src/main/java/io/camunda/zeebe/client/api/worker/JobWorkerBuilderStep1.java).
```

File: docs/product-manuals/clients/java-client-examples/job-worker-open.md [hunk 2]
```diff
@@ -15,7 +15,7 @@ title: "Open a job worker"
 
 ## JobWorkerCreator.java
 
-[Source on github](https://github.com/zeebe-io/zeebe/tree/develop/samples/src/main/java/io/zeebe/example/job/JobWorkerCreator.java)
+[Source on github](https://github.com/zeebe-io/zeebe/tree/develop/samples/src/main/java/io/camunda/zeebe/example/job/JobWorkerCreator.java)
```
Comment (korthout) [inline docs/product-manuals/clients/java-client-examples/job-worker-open.md:18]:
```suggestion
[Source on github](https://github.com/camunda-cloud/zeebe/tree/develop/samples/src/main/java/io/camunda/zeebe/example/job/JobWorkerCreator.java)
```

**Review by korthout — CHANGES_REQUESTED** [review]
Thanks for these changes, but it does not close the issue yet because not all cases are updated:

Try running `find ./docs -type f -name '*.md' | xargs egrep -i 'java/io/zeebe'` or `ack 'java/io/zeebe' ./docs` to find remaining cases. 

I've saw that you also started updating the zeebe-io organisation references to camunda-cloud. I've provided some extra ones, but there are still quite some more. Try `find ./docs -type f -name '*.md' | xargs egrep -i 'zeebe-io/zeebe/'` or `ack 'zeebe-io/zeebe/' ./docs` to find them.

**Comment by akeller** [comment]
Oh! You are totally right. I wasn't paying attention to the URLs, I was just checking whether the link 404'd and it looks like I was redirected and didn't notice 😅 .

**Comment by korthout** [comment]
Sure, no problem. Just let me know when you want me to take another look 👍

**Comment by korthout** [comment]
@akeller I think so, but I'm not entirely sure. @npepinpe Did the go import statements change as well with the package renamings?

**Comment by npepinpe** [comment]
Yes, they also changed, though note that the old ones aren't dead per se but will be redirected by Github.

**Comment by akeller** [comment]
I'll leave those alone then, thank you both!

**Comment by akeller** [comment]
I additionally updated links moved to the Community Hub like zeebe-simple-monitor.

I also updated links to raw files, like the gRPC proto buffer file - https://raw.githubusercontent.com/camunda-cloud/zeebe/develop/gateway-protocol/src/main/proto/gateway.proto

Did the inviter link change for Slack? It's still functioning as https://zeebe-slack-invite.herokuapp.com/.

Other than that, I tried to confirm the changes resolved to working URLs. Occasionally there are still some redirects, particularly when walking the directories in GitHub.

**Review by korthout — APPROVED** [review]
Thanks for the changes @akeller 🙇 

This LGTM 👏 

>Did the inviter link change for Slack? It's still functioning as https://zeebe-slack-invite.herokuapp.com/.

AFAIK, this did not change, but it might be a good idea to align naming in the future. Does not need to be part of this changeset.

> Other than that, I tried to confirm the changes resolved to working URLs. Occasionally there are still some redirects, particularly when walking the directories in GitHub.

These redirects are likely due to incorrect usage of `tree`/`blob` in the url: https://stackoverflow.com/a/61959710/3442860. This is nothing bad because GitHub handles these for us, so let's leave this out of the scope of this changeset.

**Pull request merged**